batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/dao/StudentDao.java

```diff
@@ -0,0 +1,164 @@
+import com.lftechnology.batch7crud.entity.Student;
+import com.lftechnology.batch7crud.exception.DataException;
+import com.lftechnology.batch7crud.utils.DbUtils;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A StudentDao class performs the CRUD operation in Student entity.
+ *
+ * @Author Binod Shrestha <binodshrestha@lftechnology.com>
+ * Created on 1/14/16
+ */
+public class StudentDao {
+  private static final Logger LOGGER = Logger.getLogger(StudentDao.class.getName());
+
+  public List<Student> fetch(Integer offset, Integer limit) throws DataException {
+    List<Student> studentList = new ArrayList<>();
+    ResultSet studentResult;
+
+    try (Connection conn = DbUtils.getConnection();
+            PreparedStatement ps = conn.prepareStatement(QueryConstant.FETCH_STUDENT_LIMIT_OFFSET)
+    ) {
+
+      ps.setInt(1, limit);
+      ps.setInt(2, offset);
+      studentResult = ps.executeQuery();
+
+      while (studentResult.next()) {
+        Integer id = studentResult.getInt(EntityConstant.ID);
+        Integer rollNo = studentResult.getInt(EntityConstant.ROLL);
+        String department = studentResult.getString(EntityConstant.DEPARTMENT);
+        String batch = studentResult.getString(EntityConstant.BATCH);
+        String name = studentResult.getString(EntityConstant.NAME);
+        String address = studentResult.getString(EntityConstant.ADDRESS);
+        Date dob = studentResult.getDate(EntityConstant.DOB);
+
+        Student student = new Student();
+        student.setId(id);
+        student.setRoll(rollNo);
+        student.setDepartment(department);
+        student.setBatch(batch);
+        student.setName(name);
+        student.setAddress(address);
+        student.setDob(dob);
+
+        studentList.add(student);
+      }
+
+      return studentList;
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public void insert(Student student) throws DataException {
+    try (Connection conn = DbUtils.getConnection();
+            PreparedStatement ps = conn.prepareStatement(QueryConstant.INSERT_INTO_STUDENT)
+    ) {
+
+      ps.setString(1, student.getName());
+      ps.setString(2, student.getAddress());
+      ps.setDate(3, new java.sql.Date(student.getDob().getTime()));
+      ps.setString(4, student.getDepartment());
+      ps.setString(5, student.getBatch());
+      ps.setInt(6, student.getRoll());
+      ps.executeUpdate();
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public void delete(Integer studentId) throws DataException {
+    try (Connection conn = DbUtils.getConnection();
+            PreparedStatement ps = conn.prepareStatement(QueryConstant.DELETE_FROM_STUDENT)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) "" is provided externally to the method and not sanitized before use. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2077)